lib/internal/modules/esm/module_job.js

```diff
@@ -255,5 +260,41 @@ class ModuleJob {
     return { __proto__: null, module: this.module };
   }
 }
-ObjectSetPrototypeOf(ModuleJob.prototype, null);
-module.exports = ModuleJob;
+
+// This is a fully synchronous job and does not spawn additional threads in any way.
+// All the steps are ensured to be synchronous and it throws on instantiating
+// an asynchronous graph.
+class ModuleJobSync extends ModuleJobBase {
+  constructor(loader, url, importAttributes, moduleWrap, isMain, inspectBrk) {
```

Comment:
Should this match the other constructor?

```suggestion
  constructor(loader, url, importAttributes = { __proto__: null }, moduleWrap, isMain, inspectBrk) {
```

Reply:
I don't think it should - because the arguments following it are all required. You can't use this constructor in a way that omits this argument (I also don't think you can use the other constructor without the other arguments, it looks like the default argument is a leftover that should already be removed, but this PR is not trying to refactor irrelevant paths).